changelog.d/15708.feature

```diff
@@ -0,0 +1 @@
+Add Unix Socket support for HTTP Replication Listeners. Contributed by Jason Little.
```

Comment:
```suggestion
Add Unix Socket support for HTTP listeners. Contributed by Jason Little.
```

Reply:
It is. Since this was the last part of the series, I went ahead and threw in the docs and the testing infrastructure. But the actual code changes are just for replication.
Here's the whole series:
1. #15353 
2. #15427
3. #15431
4. #15470
5. #15491
6. #15578

Additionally, adjacent and bugfixes
* #15644
* #15672
* #15746 

I'm not against changing the newsfragment, just as long as it's accurate and not to wordy(and to be fair it was written before the docs and testing stuff were added so it *does* need an update).

Reply:
```suggestion
Add Unix Socket support for HTTP Replication Listeners, with Documentation and ability to test with Complement. Contributed by Jason Little.
```
This feels to wordy, but was more what I was thinking.

Reply:
```suggestion
Add Unix Socket support for HTTP Replication Listeners. Document and provide usage instructions for utilizing Unix sockets in Synapse. Contributed by Jason Little.
```